packages/vuetify/src/components/VTextField/VTextField.ts

```diff
@@ -55,6 +55,7 @@ export default baseMixins.extend<options>().extend({
       default: '$vuetify.icons.clear',
     },
     counter: [Boolean, Number, String],
```

Comment:
Continue discussion to `counterText`:
@myleslee, @MajesticPotatoe, @jacekkarczmarczyk 

`${counterValue} / 30` vs `${counterText}``

> Maybe slot would be a good idea? You could add literally anything, including colors, icons etc. More powerful but also more to typing
> 
> ```
> <v-text-field #counter="{ value, max }">{{ someFun(value) }} / {{ max }}</v-text-field>
> ```
> 
> ```
> <v-text-field>
>   <template #counter="{ value, max }">
>     <v-icon class="red">foo</v-icon> 
>     {{ value < 10 : 'too short' : 'too long'}}, max: {{ max }}</span>
>   </template>
> </v-text-field>
> ```
> 
> Async would need an additional variable in data, but i don't think it's a frequent case.

Correct me if I'm wrong, but would a `slot` not mean a breaking change?

Reply:
The more I look at it the `slot` solution, the more uncomfortable I feel with it
It could hurt material design guidelines and is also thought to be slightly beyond the feature request 🤔

Reply:
slot would simply be a feature/enhancement. it wouldn't be breaking unless it actually broke existing functionality. In this case its just providing additional.

Reply:
IMO `counter-slot` could be another feature request and be independent of this feature